### ISSUE TITLE
论文模型结构和代码对不上

### ISSUE THREAD

**Issue by tengshaofeng** [ZHKKKe/MODNet#181]
dear ZHKKKe， 作者大大，非常崇拜你，你的线上版本的效果太好了，我想知道线上未公开的版本的网络结构是和论文里的结构一致还是和提供的代码里的结构一致啊。期待您的指导。

**Comment by ZHKKKe** [comment]
你好，感谢你的关注。
- 这个Repo中的代码和预训练模型对应于Arxiv上v1/v2版本的论文。
- 我们后来进行了一定的改进(对应Arxiv上的v3版本/AAAI版本)。改进后的代码还未更新到Repo中。
- 线上版本的模型是一个应用实践，里面包含了一些已有的但不是我们提出的技术，没有开源代码。

**Comment by tengshaofeng** [comment]
好的，非常感谢百忙中腾出时间回复，
1. 改进后的模型对应的论文地址或者论文名能否给一个呀？
2. 包含了什么技术能否大致透露以下，该技术是否在v3的论文中有体现？
3.另外我实践了一下soc，感觉对我的效果影响不大，不知道是不是自己有没有理解错

**Comment by ZHKKKe** [comment]
1. 你看的应该是[新版的文章](https://arxiv.org/pdf/2011.11961.pdf)（包含e-ASPP）
2. 包括attntion, patch-refine，semi-supervised等，是已有技术应用，在论文中没有体现。
3. SOC有一定的要求，主要包括：一定数量的，属于同一domain的无标签数据；无标签数据和标签数据domain差别不能太大。

**Comment by tengshaofeng** [comment]
@ZHKKKe 好的，感谢

**Comment by tengshaofeng** [comment]
> 1. 你看的应该是[新版的文章](https://arxiv.org/pdf/2011.11961.pdf)（包含e-ASPP）
> 2. 包括attntion, patch-refine，semi-supervised等，是已有技术应用，在论文中没有体现。
> 3. SOC有一定的要求，主要包括：一定数量的，属于同一domain的无标签数据；无标签数据和标签数据domain差别不能太大。

对了，作者大大， 我再次拜读了您的代码，貌似没看到e-aspp相应的部分，只看到se_block结构。我想问一下您的在线的版本的人像服务有这个e-aspp吗，希望得到您的指点

**Comment by jeannotes** [comment]
I have compare your papers (v1,v2,v3) carefully, and I have find that except the se-block , the others are almost the same, eg high resolution block, in your code, we find 3 interpolations, but in your papers, there are only two interpolation. @ZHKKKe @tengshaofeng

**Comment by tengshaofeng** [comment]
@jeannotes yes, you are right. In HRBranch exists 3 upsamples from enc2x, enc4x, lr8x.  
I think e-aspp is replaced by se-block.
